OpenRA.Game/Graphics/SpriteFont.cs

```diff
@@ -71,11 +75,16 @@ public void DrawText(string text, float2 location, Color c)
 				}
 
 				var g = glyphs[Pair.New(s, c)];
-				Game.Renderer.RgbaSpriteRenderer.DrawSprite(g.Sprite,
-					new float2(
-						(int)Math.Round(p.X + g.Offset.X, 0),
-						p.Y + g.Offset.Y));
+				var x = (int)Math.Round(p.X + g.Offset.X, 0);
+				var y = p.Y + g.Offset.Y;
+				Game.Renderer.RgbaSpriteRenderer.DrawSprite(g.Sprite, new float2(x ,y));
 				p.X += g.Advance;
+
+				if (face != null && face.HasKerning && i < text.Length - 1)
+				{
+					var kerning = face.GetKerning(face.GetCharIndex(s), face.GetCharIndex(text[i + 1]), KerningMode.Default);
```

Comment:
You can make this cleaner by tracking the current and last face index each time through the loop and then updating only one each time around.